packages/aragon-api-react/src/index.js

```diff
@@ -0,0 +1,150 @@
+import {
+  createContext,
+  createElement,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import Aragon, { providers } from '@aragon/api'
+import { map } from 'rxjs/operators'
+
+const postMessage = (name, value) => {
+  window.parent.postMessage({ from: 'app', name, value }, '*')
+}
+
+const requestMenu = () => {
+  postMessage('requestMenu', true)
+}
+
+const AragonApiContext = createContext()
+
+function AragonApi({
+  children,
+  reducer = state => (state === null ? {} : state),
+  onMessage = () => {},
+}) {
+  const [api, setApi] = useState(null)
+  const [connectedAccount, setConnectedAccount] = useState('')
+  const [network, setNetwork] = useState('')
+  const [appState, setAppState] = useState(null)
+  const [displayMenuButton, setDisplayMenuButton] = useState(false)
+
+  useEffect(() => {
+    setApi(new Aragon(new providers.WindowMessage(window.parent)))
+  }, [])
+
+  useEffect(() => {
+    if (AragonApi._mounted) {
+      throw new Error(
+        'AragonApi has been declared more than once. Please ensure ' +
+          'that you only have one instance in your application.'
+      )
+    }
+
+    AragonApi._mounted = true
+    return () => {
+      AragonApi._mounted = false
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!api) return
+
+    let subscribers = []
+
+    const handleMessage = ({ data }) => {
+      if (data.from !== 'wrapper') {
+        return
+      }
+      if (data.name === 'displayMenuButton') {
+        setDisplayMenuButton(data.value)
+      }
+      if (data.name === 'ready') {
+        subscribers = [
+          // app state
+          api
+            .state()
+            .pipe(map(state => reducer(state)))
+            .subscribe(state => setAppState(state)),
+
+          // account
+          api
+            .accounts()
+            .subscribe(accounts => setConnectedAccount(accounts[0] || '')),
+
+          // network
+          api.network().subscribe(network => setNetwork(network || '')),
+        ]
+
+        postMessage('ready', true)
+      }
+
+      onMessage(data.name, data.value)
+    }
+
+    window.addEventListener('message', handleMessage)
+
+    return () => {
+      window.removeEventListener('message', handleMessage)
+      subscribers.forEach(subscriber => subscriber.unsubscribe())
+    }
+  }, [api, reducer])
+
+  return createElement(
```

Comment:
Was there a particular reason to use `createElement()`? Just syntax?

Reply:
It was to avoid adding `babel-preset-react` only for one `createElement` call. Do you think we should use the JSX syntax?

Reply:
👍 Nah, but let's leave a comment for this :).